lib/events.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const P = require('./promise')
+
+const ACTIVITY_EVENTS = {
+  'account.confirmed': true,
+  'account.created': true,
+  'account.deleted': true,
+  'account.keyfetch': true,
+  'account.login': true,
+  'account.reset': true,
+  'account.signed': true,
+  'account.verified': true,
+  'device.created': true,
+  'device.deleted': true,
+  'device.updated': true
+}
+
+// We plan to emit a vast number of flow events to cover all
+// kinds of success and error steps of the sign-in/up journey.
+// It's far easier to define the events that *aren't* a flow
+// event than it is to define those that are.
+const NOT_FLOW_EVENTS = {
+  'account.deleted': true,
+  'account.reset': true,
+  'device.created': true,
+  'device.deleted': true,
+  'device.updated': true
+}
+
+const NOT_FLOW_EVENTS_IF_SERVICE_IS_CONTENT_SERVER = {
+  'account.signed': true
+}
+
+module.exports = log => {
+  return { emit: emit }
+
+  /**
+   * Emit a flow event and/or an activity event.
+   *
+   * @name emitEvent
+   * @this request
+   * @param {String} event
+   * @param {Object} [data]
+   */
+  function emit (event, data) {
+    const request = this
+
+    return P.resolve().then(() => {
+      if (ACTIVITY_EVENTS[event]) {
+        return log.activityEvent(event, request, data)
+      }
+    })
+    .then(() => {
+      if (NOT_FLOW_EVENTS[event]) {
+        return
+      }
+
+      if (
+        NOT_FLOW_EVENTS_IF_SERVICE_IS_CONTENT_SERVER[event] &&
```

Comment:
What specifically don't you like about it? Just the length or something else?

Fwiw, I wanted something named to replace the magic string that we currently use in `log.flowEvent`. And I wanted the name to be explicit about what it's for.

If you can suggest something more suitable, I'm happy to change it.